deeppavlov/models/ner/NER_model.py

```diff
@@ -0,0 +1,307 @@
+            (output_fw, output_bw), _ = tf.nn.bidirectional_dynamic_rnn(cell_fw, cell_bw, features,
+                                                                        sequence_length=self.real_sent_lengths_ph,
+                                                                        dtype=tf.float32)
+            self.output = tf.concat([output_fw, output_bw], axis=-1)
+
+            ntime_steps = tf.shape(self.output)[1]
+            self.output = tf.reshape(self.output, [-1, 2 * lstm_hidden_size])
+            layer1 = tf.nn.dropout(tf.layers.dense(inputs=self.output, units=lstm_hidden_size, activation=None,
+                                                   kernel_initializer=xavier_initializer()), self._dropout_ph)
+            pred = tf.layers.dense(inputs=layer1, units=n_tags, activation=None,
+                                   kernel_initializer=xavier_initializer())
+            self.logits = tf.reshape(pred, [-1, ntime_steps, n_tags])
+
+            log_likelihood, self.transition_params = tf.contrib.crf.crf_log_likelihood(self.logits,
+                                                                                       self._y_ph,
+                                                                                       self.real_sent_lengths_ph)
+        # loss and opt
+        with tf.variable_scope("loss_and_opt"):
+            self.loss = tf.reduce_mean(-log_likelihood)
+            self.train_op = self.get_train_op(self.loss)
+
+        self.sess = tf.Session()
+        self.sess.run(tf.global_variables_initializer())
+        self.load()
+
+    def predict(self, xs):
+        feed_dict = self._fill_feed_dict(xs)
+        logits, trans_params, sent_lengths = self.sess.run([self.logits,
+                                                            self.transition_params,
+                                                            self.real_sent_lengths_ph],
+                                                           feed_dict=feed_dict)
+        # iterate over the sentences because no batching in viterbi_decode
+        y_pred = []
+        for logit, sequence_length in zip(logits, sent_lengths):
+            logit = logit[:int(sequence_length)]  # keep only the valid steps
+            viterbi_seq, viterbi_score = tf.contrib.crf.viterbi_decode(logit, trans_params)
+            y_pred += [viterbi_seq]
+        return y_pred
+
+    def _fill_feed_dict(self, xs, y=None, train=False):
+        assert len(xs) == len(self._xs_ph_list)
+        xs = list(xs)
+        for x in xs[1:]:
+            x = np.array(x)
+        feed_dict = {ph: x for ph, x in zip(self._xs_ph_list, xs)}
+        if y is not None:
+            feed_dict[self._y_ph] = y
+        feed_dict[self.training_ph] = train
+        if not train:
+            feed_dict[self._dropout_ph] = 1.0
+
+        return feed_dict
+
+    def __call__(self, *args, **kwargs):
+        if len(args[0]) == 0 or (args[0] == [0]):
```

Comment:
What it the reason for such checking `args[0] == [0]`?

Reply:
It's used in the interaction stage to trap the lists with only one empty string.